src/test/java/com/amihaiemil/docker/ListedPluginsTestCase.java

```diff
@@ -0,0 +1,137 @@
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import org.apache.http.HttpStatus;
+import org.apache.http.NameValuePair;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.collection.IsIterableWithSize;
+import org.hamcrest.core.IsEqual;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+/**
+ * Unit tests for {@link ListedPlugins}.
+ * @author Boris Kuzmic (boris.kuzmic@gmail.com)
+ * @since 0.0.8
+ */
+public final class ListedPluginsTestCase {
+
+    /**
+     * {@link ListedPlugins} can iterate over them without
+     * filters.
+     */
+    @Test
+    public void iterateAll() {
+        final Plugins all = new ListedPlugins(
+            new AssertRequest(
+                new Response(
+                    HttpStatus.SC_OK,
+                    "[{\"Name\": \"abc1\"}, {\"Name\":\"cde2\"}]"
+                ),
+                new Condition(
+                    "iterate() must send a GET request",
+                    req -> "GET".equals(req.getRequestLine().getMethod())
+                ),
+                new Condition(
+                    "iterate() resource URL must be '/plugins'",
+                    req -> req.getRequestLine()
+                        .getUri().endsWith("/plugins")
+                )
+            ),
+            URI.create("http://localhost/plugins"),
+            Mockito.mock(Docker.class)
+        );
+        MatcherAssert.assertThat(
+            "There should be 2 plugins in the list",
+            all,
+            new IsIterableWithSize<>(
+                new IsEqual<>(2)
+            )
+        );
+        final Iterator<Plugin> itr = all.iterator();
+        MatcherAssert.assertThat(
+            "Name should match abc1",
+            itr.next().getString("Name"),
+            new IsEqual<>("abc1")
+        );
+        MatcherAssert.assertThat(
+            "Name should match cde2",
+            itr.next().getString("Name"),
+            new IsEqual<>("cde2")
+        );
+    }
+
+    /**
+     * Tests if {@link ListedPlugins} can filter plugins.
+     * @throws IOException If something goes wrong.
+     */
+    @Test
+    public void iterateWithFilters() throws IOException {
+        new ListedPlugins(
+            new AssertRequest(
+                new Response(
+                    HttpStatus.SC_OK,
+                    //@checkstyle LineLength (1 line)
+                    "[{\"Name\": \"abc1\"}, {\"Name\": \"def2\"}, {\"Name\": \"ghi3\"}, {\"Name\":\"jkl4\"}]"
+                ),
+                new Condition(
+                    "iterate() must send a GET request",
+                    req -> "GET".equals(req.getRequestLine().getMethod())
+                ),
+                new Condition(
+                    //@checkstyle LineLength (1 line)
+                    "iterate() query parameters must include the filters provided",
+                    req -> {
+                        // @checkstyle LineLength (1 line)
```

Comment:
@bkuzmic Let's rearrange this so we do not need to use this second checkstyle ignore tag